File: shoring/flywheel/src/main/java/com/panopset/flywheel/TemplateInputStream.java
package com.panopset.flywheel;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;
import com.panopset.compat.Logop;

public class TemplateInputStream extends TemplateArray {

  public TemplateInputStream(InputStream inputStream) {
    super(inputStream2list(inputStream));
  }

  private static final List<String> inputStream2list(InputStream inputStream) {
    List<String> rtn = new ArrayList<>();
    try (BufferedReader br = new BufferedReader(new InputStreamReader(inputStream))) {
      String line = br.readLine();
      while (line != null) {
        rtn.add(line);
        line = br.readLine();
      }
    } catch (IOException ex) {
      Logop.error(ex);
    }
    return rtn;
  }
}

File: code/shoring/flywheel/src/main/kotlin/com/panopset/flywheel/TemplateInputStream.kt
package com.panopset.flywheel

import com.panopset.compat.Logz
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader

class TemplateInputStream(inputStream: InputStream) :
    TemplateArray(inputStream2list( inputStream))

fun inputStream2list(inputStream: InputStream): List<String> {
    val rtn: MutableList<String> = ArrayList()
    try {
        BufferedReader(InputStreamReader(inputStream)).use { br ->
            var line = br.readLine()
            while (line != null) {
                rtn.add(line)
                line = br.readLine()
            }
        }
    } catch (ex: IOException) {
        Logz.errorEx(ex)
    }
    return rtn
}
